frontend/components/RefineButton.tsx

```diff
@@ -1,39 +1,159 @@
 import { MdSettings } from "react-icons/md";
+import { WalletContext } from "@/components/wallet/WalletContext";
+import { BurntPixArchives__factory } from "@/contracts";
+import SignInButton from "./SignInButton";
 import { inter } from "@/app/fonts";
 
 const RefineButton: React.FC = () => {
-  const buttonBgColor = "#FE005B";
-  const buttonTextColor = "white";
-  const iconButtonSize = "24px";
+  const walletContext = useContext(WalletContext);
+  const { account, provider, networkConfig } = walletContext;
+  const defaultIterations = 100;
+  const [selectedIterations, setSelectedIterations] =
+    useState(defaultIterations);
+  const [isRefining, setIsRefining] = useState(false);
+  const toast = useToast();
+  const maxIterations = 10000;
+  const defaultRed = "#FE005B";
+  const burntPixArchives = BurntPixArchives__factory.connect(
+    networkConfig.burntPixArchivesAddress,
+    provider,
+  );
 
-  return (
-    <Flex align="center" justify="right">
-      <Button
-        bg={buttonBgColor}
-        color={buttonTextColor}
-        _hover={{ bg: "red.600" }}
-        borderRadius={10}
-        h="30px"
-        w="70px"
-        fontSize="md"
-        fontFamily={inter.style.fontFamily}
-        fontWeight={700}
-      >
-        REFINE
-      </Button>
+  const refine = async () => {
+    if (isRefining) return;
+    setIsRefining(true);
+    try {
+      console.log("Refining with iterations: ", selectedIterations);
+      const signer = await provider.getSigner();
+      await burntPixArchives
+        .connect(signer)
+        ["refineToArchive(uint256)"](selectedIterations);
+      setIsRefining(false);
+    } catch (error: any) {
+      setIsRefining(false);
+      let message = "Error refining: ";
+      if (error.code === "ACTION_REJECTED") {
+        message += "Transaction rejected.";
+      } else if (error.action === "estimateGas") {
+        message +=
+          "Error estimating gas, try with a lower number of iterations.";
+      } else {
+        message += error.message;
+      }
+      toast({
+        title: message,
+        status: "error",
+        position: "bottom-left",
+        duration: 5000,
+        isClosable: true,
+      });
+    }
+  };
 
-      <Button
-        size="sm"
-        variant="ghost"
-        color={buttonBgColor}
-        _hover={{ color: "red.600" }}
-        p={0}
-      >
-        <Icon as={MdSettings} boxSize={iconButtonSize} />
-      </Button>
-    </Flex>
+  return (
+    <Box>
+      {account ? (
+        <Flex align="center" justify="right">
+          <Button
+            bg={defaultRed}
+            color="white"
+            _hover={{ bg: defaultRed }}
+            borderRadius={10}
+            h="30px"
+            w="fit-content"
+            fontSize="md"
+            fontWeight={700}
+            onClick={refine}
+            fontFamily={inter.style.fontFamily}
+          >
+            {isRefining ? "REFINING..." : "REFINE TO ARCHIVE"}
+          </Button>
```

Comment:
```suggestion
          <Button
            bg={defaultRed}
            color="white"
            _hover={{ bg: defaultRed }}
            borderRadius={10}
            h="30px"
            w="fit-content"
            fontSize="md"
            fontWeight={700}
            onClick={refine}
            fontFamily={inter.style.fontFamily}
            loadingText={"REFINING..."}
            isLoading={isRefining}
          >
            REFINE TO ARCHIVE
          </Button>
```


If we use loadingText/isLoading we get the benefit of a disabled state

<img width="268" alt="Screenshot 2024-04-05 at 14 01 58" src="https://github.com/yearone-io/burntpix-archives/assets/132473036/dc100ecb-b053-4247-b680-3175409bbdb9">

Reply:
nice thinking @democrazyfi. @Ledespana worth trying the above imo as well.